api/v3/utils.php

```diff
@@ -1235,7 +1235,7 @@ function formatCheckBoxField(&$checkboxFieldValue, $customFieldLabel, $entity) {
  * @return array
  */
 function _civicrm_api3_basic_get($bao_name, $params, $returnAsSuccess = TRUE, $entity = "", $sql = NULL, $uniqueFields = FALSE) {
-  $entity = CRM_Core_DAO_AllCoreTables::getBriefName(str_replace('_BAO_', '_DAO_', $bao_name));
+  $entity = $entity ?: CRM_Core_DAO_AllCoreTables::getBriefName(str_replace('_BAO_', '_DAO_', $bao_name));
```

Comment:
@eileenmcnaughton can you correct this to
```$entity = $entity ? $entity : CRM_Core_DAO_AllCoreTables::getBriefName(str_replace('_BAO_', '_DAO_', $bao_name));```

Reply:
@monishdeb that syntax is actually a correct variant on the tenary

"Since PHP 5.3, it is possible to leave out the middle part of the ternary operator. Expression expr1 ?: expr3 returns expr1 if expr1 evaluates to TRUE, and expr3 otherwise."

http://php.net/manual/en/language.operators.comparison.php

Reply:
(in later php you can use var that might not be set that way which saves a lot of isset())

Reply:
Ahh right, didn't know that. Thanks for explaining it to me :)